element-hq/element-x-android
# Format RoomListRoomSummary timestamps in real time

## Content

- Make `RoomListRoomSummary.timestamp` a `Date?` instead of a `String?`.
- Add `RoomListContentState.Rooms.formatTimestamp` and `RoomListState.formatTimestamp` lambdas: these lambdas use `RoomLastMessageFormatter` under the hood to format any provided dates in an inexpensive way, and the same instance is reused, avoiding unnecessary expensive instantiations.
- Modify the timestamps provided for the previews.

## Motivation and context

There is a bug where in the room list you might be able to see rooms whose state hasn't changed since yesterday still displaying the time at which some event happened instead of 'Yesterday', making you think they happened at some time today and they're out of order. 

If you open one of these rooms, the room list item is also refreshed and the right 'Yesterday' formatted timestamp is used.

## Screenshots / GIFs

There are a few changes in the previews for room list items. I've tried to keep at least some with the same values.

## Tests

- Open the app until the room list loads.
- Change the system's date in the device settings to some other date or tomorrow, i.e.
- Go back to the app and check the timestamps have changed.

## Tested devices

- OS version(s):

## Checklist
